docs/proposals/implemented/2017-08-split-plans-from-serviceclass.md

```diff
@@ -0,0 +1,124 @@
+# Split Plans from ServiceClasses
+
+Proposed: 2017-08-08
+Approved: 2017-08-15
+
+## Abstract
+
+Proposal to split Plans from ServiceClass to become a top-level object in our API.
+
+## Motivation
+
+If we get a ServiceClass, we get all of the Plans, which potentially contain a
+lot of data. The data of concern is primarily the Schema information used for
+exposing the details of a ServicePlan to a UI.
+
+## Constraints and Assumptions
+
+ - It is desired to have the ServicePlan data in one place and not
+   duplicated. Duplicated data gets out of sync.
+
+## Proposed Design
+
+ - new ServicePlan resource based on existing ServicePlan object
+ - Implement FieldSelector for ServicePlan on `Free` field combined with ServiceClass Reference
+
+### ServiceClass API Changes
+
+Instead of having an array of ServicePlans defined inline, we have an array of
+Reference to ServicePlan.
```

Comment:
Isn't that reverse, `ServicePlan` having a reference to a `ServiceClass`? Do we still have an array for some reason?

Reply:
Good catch. I will edit this and review the whole thing to make sure it is correct.

Reply:
brain fart on my part. I will update and review the whole document.